### PR TITLE
Rework MacOS documentation

### PR DESCRIPTION
**Pull request by nikita-bykov**
# Description
Rework MacOS documentation to be consistent with Ubuntu and Windows tests

#### Related issue:
[Rework MacOS documentation to be consistent with Ubuntu and Windows tests #1604](https://github.com/actions/virtual-environments-internal/issues/1604)
## Check list
- [x] Related issue / work item is attached
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

## Builds: 
[macOS-10.14](https://github.visualstudio.com/virtual-environments/_build/results?buildId=93586&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb&t=0cc083f2-e4b4-5c2b-a1d7-966043ae867c), [macOS-10.15](https://github.visualstudio.com/virtual-environments/_build/results?buildId=93587&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb&t=0cc083f2-e4b4-5c2b-a1d7-966043ae867c), [macOS-11.0](https://github.visualstudio.com/virtual-environments/_build/results?buildId=93584&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb&t=0cc083f2-e4b4-5c2b-a1d7-966043ae867c)

### CODE REVIEW

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 1]
```diff
@@ -129,4 +130,320 @@ function Build-OSInfoSection {
...
+    $mongo = Run-Command "mongo --version" | Select-String "MongoDB shell version" | Take-Part -Part 3
+    return "mongo $mongo"
+}
+
+function Get-MongodVersion {
+    $mongod = Run-Command "mongod --version" | Select-String "db version " | Take-Part -Part 2
+    return "mongod $mongod"
+}
+
+function Get-P7zipVersion {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Common.psm1:341]: `P7zipVersion` typo?

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 2]
```diff
@@ -129,4 +130,320 @@ function Build-OSInfoSection {
...
+    $gnuTar = Run-Command "gtar --version" | Select-String "tar" | Take-Part -Part 3
+    return "GNU Tar $gnuTar - available by 'gtar' alias"
+}
+
+function Get-BsdtarVersion {
+    $bsdtar = Run-Command "tar --version" | Take-Part -Part 1
+    return "bsdtar $bsdtar - available by 'tar' alias"
+}
+
+function Get-NemanVersion {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Common.psm1:356]: `Neman` typo

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 3]
```diff
@@ -129,4 +130,320 @@ function Build-OSInfoSection {
...
+    $newmanVersion = Run-Command "newman --version"
+    return "Newman $newmanVersion"
+}
+
+function Get-VboxVersion {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Common.psm1:361]: Get-VirtualBoxVersion

File: images/macos/software-report/SoftwareReport.Xcode.psm1 [hunk 4]
```diff
@@ -76,6 +76,11 @@ function Get-XcodePlatformOrder {
     }
 }
 
+function Get-XcodeVersion {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Xcode.psm1:79]: `Get-XcodeCommandLineToolsVersion`

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 1 above)
Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Fixed

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 3 above)
Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Fixed

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 2 above)
Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Fixed

File: images/macos/software-report/SoftwareReport.Xcode.psm1 (hunk 4 above)
Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Xcode.psm1]: Fixed

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 5]
```diff
@@ -53,187 +42,123 @@ if ($os.IsLessThanBigSur) {
...
-    "AWS SAM CLI ${awsSamVersion}",
-    "AWS Session Manager CLI ${awsSessionManagerVersion}",
-    "Aliyun CLI ${aliyunVersion}",
-    "Xcode Command Line Tools ${xcodeVersion}"
+    (Get-FastlaneVersion),
+    (Get-CmakeVersion),
+    (Get-AppCenterCLIVersion),
+    (Get-AzureCLIVersion),
+    (Get-AWSVersion),
+    (Get-AWSSAMVersion),
```
Comment (AlenaSviridenko) [inline images/macos/software-report/SoftwareReport.Generator.ps1:138]: nitpick: `AWSCLIVersion` and `AWSSAMCLIVersion` for consistency with other tools naming

**Comment by AlenaSviridenko** [comment]
Some names are missing
image

image

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 6]
```diff
@@ -129,4 +130,320 @@ function Build-OSInfoSection {
...
+}
+
+function Get-NuGetVersion {
+    $nugetVersion = Run-Command "nuget help" | Select-Object -First 1 | Take-Part -Part 2
+    return "NuGet $nugetVersion"
+}
+
+function Get-CondaVersion {
+    $condaVersion = Invoke-Expression "conda --version"
+    return "Mini $condaVersion"
```
Comment (AlenaSviridenko) [inline images/macos/software-report/SoftwareReport.Common.psm1:207]: Miniconda without space
image

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 5 above)
Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Generator.ps1]: Renamed these functions

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 6 above)
Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Thank you, fixed it

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 7]
```diff
@@ -129,4 +130,320 @@ function Build-OSInfoSection {
...
+}
+
+function Get-PHPVersion {
+    $PHPVersion = Run-Command "php --version" | Select-Object -First 1 | Take-Part -Part 0,1
+    return $PHPVersion
+}
+
+function Get-NodeVersion {
+    $nodeVersion = Run-Command "node --version"
+    return "Node $nodeVersion"
```
Comment (AlenaSviridenko) [inline images/macos/software-report/SoftwareReport.Common.psm1:142]: It should be `Node.js`, not just `Node`

Comment (nikita-bykov) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Renamed it

**Review by AlenaSviridenko — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by MFredrich — COMMENTED** [review]
Duplicate of #
images/macos/software-report/SoftwareReport.Common.psm1

**Pull request merged**